packages/x-components/tests/e2e/cucumber/recommendations/recommendations.spec.ts

```diff
@@ -1,22 +1,44 @@
 import { And, Given, Then } from 'cypress-cucumber-preprocessor/steps';
-import { InstallXOptions } from '../../../src/x-installer/x-installer/types';
+import { InstallXOptions } from '../../../../src/x-installer/x-installer/types';
+
+let config: InstallXOptions['xModules'];
 
 Given('following config: max items to store is {int}', (maxItemsToRequest: number) => {
-  const config: InstallXOptions['xModules'] = {
+  config = {
     recommendations: {
       config: {
         maxItemsToRequest
       }
     }
   };
-  cy.visit('/test/recommendations', {
+});
+
+And('suggestions response being mock {string}', (mockName: string) => {
```

Comment:
Main changes from the proposed solution in the task's comment are here:

* The `intercept` has to be done before the request is done so it has to be set up before the `visit call`.
* Dynamic import is not that dynamic after all and you can't use a variable string as the first argument. My proposed solution is to have just one file for the stubs (instead of one per stub as suggested) and dynamically select the mock to use from the ones being exported in the stub file.

Reply:
I noticed that the dynamic import with dynamic argument doesn't work because it's not resolved because the spec are not pre processed correctly or in the correct order. https://docs.cypress.io/guides/tooling/plugins-guide.html#Preprocessors. We can leave it as it's but just a comment just in case we need it in the future.

Reply:
Leaving it as-is for now at least as I also prefer one stub file if possible.

Reply:
This issue is in reality caused by webpack, probably some config issues in vue-cli. I've to look deeper into it.

References:
https://github.com/webpack/webpack/issues/6680
https://webpack.js.org/api/module-methods/#dynamic-expressions-in-import